src/Grpc.ChannelConfiguration/GrpcServiceMethodConfigurationExtensions.cs

```diff
@@ -0,0 +1,281 @@
+﻿using Grpc.Net.Client.Configuration;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Extensions.DependencyInjection;
+
+/// <summary>
+/// A set of extension methods for configuring an <see cref="IHttpClientBuilder"/>.
+/// </summary>
+public static class GrpcServiceMethodConfigurationExtensions
+{
+    /// <summary>
+    /// The name of the configuration key to be used for the fallback setup.
+    /// </summary>
+    public const string FallbackConfigurationKeyName = "Default";
+
+    /// <summary>
+    /// Customizes the <see cref="IHttpClientBuilder" /> to configure all service methods according to the values in <paramref name="section"/>.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHttpClientBuilder" /> to configure.</param>
+    /// <param name="section">The instance of <see cref="IConfigurationSection" /> containing a child for each service method to be configured.</param>
+    /// <returns>The configured <see cref="IHttpClientBuilder" />.</returns>
+    /// <exception cref="ArgumentException">Thrown when any method is configured with both a <see cref="RetryPolicy" /> and a <see creft="HedgingPolicy" />.</exception>
+    public static IHttpClientBuilder ConfigureServiceMethods(this IHttpClientBuilder builder, IConfigurationSection section)
+    {
+        _ = builder ?? throw new ArgumentNullException(nameof(builder));
+
+        _ = section ?? throw new ArgumentNullException(nameof(section));
+
+        foreach (var item in section?.GetChildren() ?? Array.Empty<IConfigurationSection>())
```

Comment:
I would split this line into two and define a variable for readability's sake
```suggestion
        var children = section?.GetChildren() ?? Array.Empty<IConfigurationSection>();
        foreach (var item in children)
```